### PR TITLE
Remove Popover from LinkControl component

### PR DESCRIPTION
**Pull request by youknowriad**
We should be able to use LinkControl outside of Popover.

**Testing instructions**

 - Check that the buttons and navigation block still work as intended. (url assigning)

### CODE REVIEW

File: packages/block-library/src/navigation-link/edit.js [hunk 1]
```diff
@@ -164,21 +165,23 @@ function NavigationLinkEdit( {
...
-								opensInNewTab: newOpensInNewTab,
-							} = {} ) => setAttributes( {
-								title: escape( newTitle ),
-								url: newURL,
-								label: label || escape( newTitle ),
-								opensInNewTab: newOpensInNewTab,
-							} ) }
-							onClose={ () => setIsLinkOpen( false ) }
-						/>
+						<Popover>
```
Comment (ellatrix) [inline packages/block-library/src/navigation-link/edit.js]: Does it not need a position here?

**Review by ellatrix — APPROVED** [review]
Sounds good
Checked with https://github.com/WordPress/gutenberg/pull/19638/files?utf8=✓&diff=unified&w=1

File: packages/block-editor/src/components/link-control/index.js [hunk 2]
```diff
@@ -176,77 +164,55 @@ function LinkControl( {
 	};
 
 	return (
-		<Popover
-			className={ classnames( 'block-editor-link-control', className ) }
-			onClose={ closeLinkUI }
-			position="bottom center"
-			focusOnMount="firstElement"
```
Comment (ellatrix) [inline packages/block-editor/src/components/link-control/index.js]: Are these props no longer needed?

File: packages/block-library/src/navigation-link/edit.js (hunk 1 above)
Comment (youknowriad) [reply packages/block-library/src/navigation-link/edit.js]: It seemed to work without it, maybe it was the default, I'll check.

File: packages/block-editor/src/components/link-control/index.js [hunk 3]
```diff
@@ -176,77 +159,66 @@ function LinkControl( {
...
 }
 
-export default compose(
-	withInstanceId,
-	withSelect( ( select, ownProps ) => {
-		if ( ownProps.fetchSearchSuggestions && isFunction( ownProps.fetchSearchSuggestions ) ) {
-			return;
-		}
-
+function ConnectedLinkControl( props ) {
```
Comment (youknowriad) [inline packages/block-editor/src/components/link-control/index.js]: I don't like that I had to do that to make unit tests pass (test the unconnected component). Ideally, we'd find a way to easily mock useSelect

File: packages/block-library/src/navigation-link/edit.js (hunk 1 above)
Comment (youknowriad) [reply packages/block-library/src/navigation-link/edit.js]: I restored this

File: packages/block-editor/src/components/link-control/index.js (hunk 2 above)
Comment (youknowriad) [reply packages/block-editor/src/components/link-control/index.js]: They seem unnecessary in my testing.

File: packages/block-editor/src/components/link-control/index.js (hunk 3 above)
Comment (aduth) [reply packages/block-editor/src/components/link-control/index.js]: > I don't like that I had to do that to make unit tests pass (test the unconnected component). Ideally, we'd find a way to easily mock useSelect

Follow-up at #19705

File: packages/block-library/src/navigation-link/edit.js [hunk 4]
```diff
@@ -164,21 +165,23 @@ function NavigationLinkEdit( {
...
+									title: newTitle = '',
+									url: newURL = '',
+									opensInNewTab: newOpensInNewTab,
+								} = {} ) => setAttributes( {
+									title: escape( newTitle ),
+									url: newURL,
+									label: label || escape( newTitle ),
+									opensInNewTab: newOpensInNewTab,
+								} ) }
+								onClose={ () => setIsLinkOpen( false ) }
```
Comment (aduth) [inline packages/block-library/src/navigation-link/edit.js]: This should have been moved to the `Popover`. `LinkControl` no longer has an `onClose` prop with the changes from this pull request.

Fix at: #19885

**Pull request merged**